crates/swc_ecma_parser/src/parser/stmt/module_item.rs

```diff
@@ -153,8 +153,16 @@ impl<'a, I: Tokens> Parser<I> {
         let _ = cur!(self, false);
         let asserts = if self.input.syntax().import_assertions()
             && !self.input.had_line_break_before_cur()
-            && eat!(self, "assert")
+            && is!(self, "assert")
         {
+            let valid = self.target() >= EsVersion::Es2022;
```

Comment:
Also, checking for `es2022` is also wrong. Import assertions is not `es2022`. It's stage 3, not 4.

Reply:
I'm not sure they should be stripped out, since host environments can fail an import without assertions that would succeed with them, as happens with JSON modules in HTML's module loader algorithm. But I guess this doesn't have to be the case generally, so stripping them might be fine, since it's not like the codegen will result in code that fails at runtime.

Reply:
> Also, checking for `es2022` is also wrong. Import assertions is not `es2022`. It's stage 3, not 4.

There's no variant for stage 3 or "ESNext" in EsVersion. I went with ES2022 because with it shipping in browsers now, it's extremely likely that it'll be bumped to stage 4 in the next few months. But what should I use instead?

Reply:
Maybe an experimental option I think?